app/src/main/java/org/thoughtcrime/securesms/conversation/v2/messages/QuoteView.kt

```diff
@@ -78,11 +78,13 @@ class QuoteView : LinearLayout {
         }
         val body = quoteViewBodyTextView.text
         val bodyTextViewIntrinsicHeight = TextUtilities.getIntrinsicHeight(body, quoteViewBodyTextView.paint, maxContentWidth)
+        val staticLayout = TextUtilities.getIntrinsicLayout(body, quoteViewBodyTextView.paint, maxContentWidth)
         result += bodyTextViewIntrinsicHeight
         if (!quoteViewAuthorTextView.isVisible) {
-            // We want to at least be as high as the cancel button, and no higher than 56 DP (that's
-            // approximately the height of 3 lines.
-            return min(max(result, toPx(32, resources)), toPx(56, resources))
+            // We want to at least be as high as the cancel button 36DP, and no higher than 3 lines of text.
+            // Height from intrinsic layout is the height of the text before truncation so we shorten
+            // proportionally to our max lines setting.
+            return max(toPx(32, resources) ,min((result / staticLayout.lineCount) * 3, result))
```

Comment:
Is it possible to run me through the second half of this, `min((result / staticLayout.lineCount) * 3, result))`?